dialect/pgdialect/array.go

```diff
@@ -121,18 +125,33 @@ func (d *Dialect) arrayAppender(typ reflect.Type) schema.AppenderFunc {
 			v = v.Elem()
 		}
 
-		b = append(b, "'{"...)
+		if kind == reflect.Slice {
+			elemType := v.Type().Elem()
+
+			if !elemType.Implements(driverValuerType) {
+				for elemType.Kind() == reflect.Ptr {
+					elemType = elemType.Elem()
+				}
+
+				if elemType.Kind() == reflect.Struct {
+					brackets = "[]"
```

Comment:
Is this meant to handle JSONB? But why would the JSONB type end up being processed here?

Reply:
These are needed to format a JSON array since the underlying column is just JSONB.  

I think I need to add a check here to ignore this if the type is no JSONB or if the elemType implement query appender to handle your example above correctly.  Will add that test and fix.